rollup.config.js

```diff
@@ -3,9 +3,12 @@ import sourcemaps from 'rollup-plugin-sourcemaps';
 import {plugins} from './build/rollup_plugins';
 
 const version = JSON.parse(fs.readFileSync('package.json')).version;
-
-const production = process.env.BUILD === 'production';
-const outputFile = production ? 'dist/mapbox-gl.js' : 'dist/mapbox-gl-dev.js';
+const {BUILD, MINIFY} = process.env;
+const minified = MINIFY === 'true';
+const production = BUILD === 'production';
+const outputFile =
+    production && minified ? 'dist/mapbox-gl.js' :
+    production ? 'dist/mapbox-gl-unminified.js' : 'dist/mapbox-gl-dev.js';
```

Comment:
This still doesn't account for the matrix of `BUILD`/`MINIFY` yet :) How about
```js
const suffix = (production ? '' : '-dev') + (minified ? '' : '-unminified');
const outputFile = 'dist/mapbox-gl' + suffix + '.js';
```

Reply:
@kkaefer I wouldn't want to change names of existing builds. This suggestion would rename `mapbox-gl-dev.js` to `mapbox-gl-dev-unminified.js`.

Reply:
Let's reverse the MINIFY flag and default to true, then specify `MINIFY:false` for the non-minified production build

Reply:
@kkaefer and rename the current `mapbox-gl.js` build to `mapbox-gl-min.js`?

Reply:
No, the suffix only gets added for unminified builds

Reply:
@kkaefer so it would still be `mapbox-gl-dev-unminified.js`? I'm confused.

Reply:
Sorry for the confusion; I didn't realize that the dev version is unminified. What I was trying to get at is that we have 4 possible combinations of `BUILD` and `MINIFY`, but we're only producing 3 different file names; the one for `BUILD:development,MINIFY:true` produces the same output file.

Reply:
Since it doesn't make sense to have a minified dev build anyway, I don't think we need to worry about it. I added another commit that hopefully makes build naming logic clearer. Feel free to add any changes you feel are needed here and I'll review, and otherwise we can merge.